fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/BoundFilter.java

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.druid.model.filter;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import javax.validation.constraints.Null;
+import java.util.Objects;
+
+/**
+ * Bound filters supports filtering on ranges of dimension values.
+ */
+public class BoundFilter extends DimensionalFilter<BoundFilter> {
+
+    /**
+     * Ordering specified for the range filtering.
+     */
+    public enum Ordering {
+        LEXICOGRAPHIC,
+        ALPHANUMERIC,
+        NUMERIC,
+        STRLEN;
+
+        final String orderingType;
+
+        /**
+         * Constructor.
+         */
+        Ordering() {
+            this.orderingType = EnumUtils.enumJsonName(this);
+        }
+
+        /**
+         * Get the JSON representation of this class.
+         *
+         * @return the JSON representation.
+         */
+        @JsonValue
+        public String toJson() {
+            return orderingType;
+        }
+    }
+
+    private final String lower;
+    private final String upper;
+    private final Boolean lowerStrict;
+    private final Boolean upperStrict;
+    private final Ordering ordering;
+
+    /**
+     * Constructor.
+     *
+     * @param dimension The druid dimension to be filtered
+     * @param lower The lower bound of the dimension value to be filtered (Optional)
+     * @param upper The upper bound of the dimension value to be filtered (Optional)
+     * @param lowerStrict Boolean to enable/ disable strict filtering for lower bounds (Optional)
+     * @param upperStrict Boolean to enable/ disable strict filtering for upper bounds (Optional)
+     * @param ordering The Ordering to be applied for the dimension filtering
+     */
+    public BoundFilter (
+            Dimension dimension,
+            @Null String lower,
+            @Null String upper,
+            @Null Boolean lowerStrict,
+            @Null Boolean upperStrict,
```

Comment:
We prefer prefer primitives to boxed primitives for the following reasons:

* Primitives are generally more time- and space-efficient than boxed primitives
* It avoids the mis-use of `==` operator on boxed primitives
* When primitives and boxed primitives are mixed in a single operation, the boxed primitive is auto-unboxed, resulting in the potential `NullPointerException` runtime error
* Instantiating boxed primitives is slower

The legitimate use cases for boxed primitives in Fili is when

1. they are used as elements, keys, and values in collections, because you cannot put primitives in collections.
2. they are used as type parameters in parameterized types
3. making reflective method invocations